src/ray/raylet/node_manager.cc

```diff
@@ -2213,6 +2226,40 @@ std::string NodeManager::DebugString() const {
   return result.str();
 }
 
+void NodeManager::RecordMetrics() const {
```

Comment:
Who will call this function?
Also, please make sure this function doesn't execute when metrics isn't enabled.

Reply:
I think we shouldn't consider whether disable metrics here. The reasons are:
1) We have already consider this when recording.
2) This is one of the way we record, and we also record metrics other ways.

Reply:
But this function's cost might not be trivial, considering the loops and functions calls. So we do a `if(not_enabled) return;` here